presto-docs/src/main/sphinx/security/hive.rst

```diff
@@ -0,0 +1,241 @@
+file using the following properties:
+
+================================================== ============================================================ ==========
+Property Name                                      Description                                                  Default
+================================================== ============================================================ ==========
+``hive.hdfs.authentication.type``                  One of ``SIMPLE`` or ``KERBEROS``. When set to ``KERBEROS``, ``SIMPLE``
+                                                   the Hive connector authenticates to HDFS using Kerberos.
+                                                   When using the default value of ``SIMPLE``, Kerberos
+                                                   authentication is disabled and no other properties need to
+                                                   be configured.
+
+``hive.hdfs.presto.principal``                     The Kerberos principal that Presto will use when connecting
+                                                   to HDFS.
+
+``hive.hdfs.presto.keytab``                        The path to the keytab file that contains a key for the
+                                                   principal specified by ``hive.hdfs.presto.principal``.
+
+================================================== ============================================================ ==========
+
+.. _hive-security-simple:
+
+``SIMPLE``
+^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=SIMPLE
+
+``SIMPLE`` authentication is the default authentication mechanism for HDFS.
+When using ``SIMPLE`` authentication, Presto connects to HDFS using Hadoop's
+simple authentication mechanism. Kerberos is not used.
+
+.. _hive-security-kerberos:
+
+``KERBEROS``
+^^^^^^^^^^^^
+
+.. code-block:: none
+
+    hive.hdfs.authentication.type=KERBEROS
+    hive.hdfs.presto.principal=hdfs@EXAMPLE.COM
+    hive.hdfs.presto.keytab=/etc/presto/hdfs.keytab
+
+When using ``KERBEROS`` authentication, Presto accesses HDFS as the principal
+specified by the ``hive.hdfs.presto.principal`` property. Presto will
+authenticate this principal using the keytab specified by the
+``hive.hdfs.presto.keytab`` keytab.
+
+Keytab files must be distributed to every node in the cluster that runs Presto.
+
+:ref:`Additional information on keytab files.<hive-security-additional-keytab>`
+
+.. _hive-security-impersonation:
+
+End User Impersonation
+======================
+
+Impersonation Accessing HDFS
+----------------------------
+
+Presto can impersonate the end user who is running a query. In the case of a
+user running a query from the command line interface, this is the value of the
+username associated with the Presto cli process or the value of the optional
+``--user`` option.  Impersonating the end user can provide additional security
+when accessing HDFS if HDFS permissions or ACLs are used.
```

Comment:
HDFS permissions are always present, and they can be set to allow varying levels of access. HDFS ACLs are optional, and can be used give more fine-grained access control than you can achieve with just the permissions. In both cases, these are features of HDFS, and aren't intrinsically tied to Kerberos.

If you aren't using impersonation, whatever user Presto is running as needs to be able to access the data files on HDFS. In practice, this would mean either setting the permissions in such a way that this is allowed, or adding an ACL that grants permission.

Some customers set up more fine-grained access control using ACLs such that Eric can read some tables, Brenda can read some tables, and neither of them can read each other's tables. Also, there's a table they can both read.

In order for these to be useful when accessing the data through Presto, Presto needs to impersonate the end user. If you don't do that, one of the following will happen:
1. The user presto is running as has too much permission and Eric can see Brenda's data and vice-versa.
2. The user presto is running as has too little permission, and nobody can see any data.
3. Some combination of the above if different tables are set up differently. Everybody is annoyed because they have no idea what they can or should be able to access.

The connection to Kerberos is that Kerberos ensures that when I claim to be Eric when I log in, I present evidence to Presto that I actually am Eric. Without Kerberos, Presto is trusting something else to ensure that Eric is really Eric.